src/cli/cli.cpp

```diff
@@ -1898,6 +1898,9 @@ void Interpreter::ProcessLine(char *aBuf, uint16_t aBufLength, Server &aServer)
         {
             argv[argc++] = cmd;
         }
+
+        VerifyOrExit(argc <= kMaxArgs,
```

Comment:
This does not do what is intended.  The loop could iterate again and write to `argv[kMaxArgs]`.